platform/src/main/java/org/hillview/sketches/Centroids.java

```diff
@@ -0,0 +1,105 @@
+package org.hillview.sketches;
+
+import org.hillview.table.api.IColumn;
+import org.hillview.table.api.IMembershipSet;
+import org.hillview.table.api.IRowIterator;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * This class represents a set of centroids of a partitioning of a table. The number of centroids should not be too
+ * large, as they are shipped over the network. Since row/column entries are only counted if the row is present in
+ * the column, it essentially computes the mean of each column over only the non-missing entries.
+ */
+public class Centroids<T> implements Serializable {
+    /**
+     * Map from the partition key to the sum of the values in the partition in every column.
+     */
+    public HashMap<T, double[]> sums;
+    /**
+     * Map from the partition key to the count of values in every column of that partition.
+     */
+    public HashMap<T, long[]> counts;
+
+    /**
+     * Constructs a zero-centroids.
+     */
+    public Centroids() {
+        this.sums = new HashMap<T, double[]>();
+        this.counts = new HashMap<T, long[]>();
+    }
+
+    /**
+     * Construct a centroids object that has processed the sums of the partitions in 'table'.
```

Comment:
Please describe what happens for missing values.
Your treatment of missing values seems to be good.